### PR TITLE
REF: Enhance consistency on Styler methods

### PR DESCRIPTION
**Pull request by staticdev**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- Added an entry in the latest `doc/source/whatsnew/v1.4.1.rst` file if fixing a bug or adding a new feature. NOT NEEDED

pandas 1.4.0 introduces deprecation of `hide_columns()` and `hide_index()` in favor to `hide()`. Problem is the messages docstrings, parameters and warning messages pass parameters in different ways making it confusing and non-readable. This PR was created to address that and make all those consistent.

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -2551,16 +2551,16 @@ def hide_columns(
...
             stacklevel=find_stack_level(),
         )
-        return self.hide(axis=1, level=level, subset=subset, names=names)
+        return self.hide(axis="columns", level=level, subset=subset, names=names)
 
     def hide(
         self,
         subset: Subset | None = None,
-        axis: Axis = 0,
+        axis: Axis = "index",
```
Comment (attack68) [inline pandas/io/formats/style.py:2563]:
```suggestion
        axis: Axis = 0,
```
most methods with str and int input default to 0. leave as is for consistency.

**Comment by attack68** [comment]
This effectively just changes the quotation symbol from single to double quotes in a warning. No need for a whatsnew entry.

Not really convinced that the error is unreadable with single quotes instead of double:

Screen Shot 2022-02-02 at 19 08 20

File: pandas/io/formats/style.py (hunk 1 above)
Comment (staticdev) [reply pandas/io/formats/style.py:2563]: Most of my colleagues I know use pandas agree that using Axis as string greatly improves readability specially to those that work with a lot of code from other libraries. This notation was introduces later and this is why is more frequent on the code. I would counter-propose, if you agree, just to change the only other method in this module that initializes Axis = 0, which is `set_sticky()`.

Comment (attack68) [reply pandas/io/formats/style.py:2563]: if `set_sticky` is not default `Axis=0` then it should be, we can make that a DOC change.

Of course your own code can include `axis="columns"` etc., that's why it's available. But changing the pandas internal code will have no impact on external code readability.

**Comment by jreback** [comment]
> This effectively just changes the quotation symbol from single to double quotes in a warning. No need for a whatsnew entry.
> 
> Not really convinced that the error is unreadable with single quotes instead of double:
> 
> Screen Shot 2022-02-02 at 19 08 20

can you check this for consistency with other deprecations that include quotes. a) we are probably not consistent b) we could check forth this (maybe)

**Comment by staticdev** [comment]
@attack68 @jreback I checked on whole code, warning is consistent with the quotes inside the columns, also the command is also consistent with black if people want to copy-paste the code.

File: pandas/io/formats/style.py (hunk 1 above)
Comment (staticdev) [reply pandas/io/formats/style.py:2563]: You are right, it doesn't improve external code, but it does improve internal code readability. This is what this PR is all about, making pandas code better for users who want to dig in debugging and also to make it easier to contribute and make it even more awesome.

**Comment by jreback** [comment]
@staticdev status of this?

**Comment by staticdev** [comment]
@jreback from my side it is ready to be merged.

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -2230,7 +2230,7 @@ def set_caption(self, caption: str | tuple) -> Styler:
 
     def set_sticky(
         self,
-        axis: Axis = 0,
+        axis: Axis = "index",
```
Comment (attack68) [inline pandas/io/formats/style.py:2233]:
```suggestion
        axis: Axis = 0,
```

as previously mentioned pandas defaults Axis to integers

File: pandas/io/formats/style.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style.py:2648]: pandas defaults Axis to integers in general.

**Comment by staticdev** [comment]
@attack68 fine, I commited your suggestion.

**Review by attack68 — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @staticdev and @attack68 for merging in main one more time.

**Pull request merged**